libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -130,8 +130,11 @@ class ClientContext
         }
         _connect_pending = 1;
         _op_start_time = millis();
-        // This delay will be interrupted by esp_schedule in the connect callback
-        delay(_timeout_ms);
+        // Following delay will be interrupted by connect callback
+        for (decltype(_timeout_ms) i = 0; _connect_pending && i < _timeout_ms; i++) {
+               // let a chance to recurrent scheduled functions (ethernet)
```

Comment:
Wording: Give scheduled functions a chance to run (e.g. Ethernet uses recurrent)